core/src/main/java/com/oath/oak/OakRValueBufferImpl.java

```diff
@@ -198,12 +196,31 @@ public double getDouble(int index) {
     }
 
     @Override
-    public void unsafeCopyBufferToIntArray(int srcPosition, int[] dstArray, int countInts) {
-        Slice s = getValueSlice();
-        start(s);
-        ByteBuffer dup = valueOperator.getValueByteBufferNoHeader(s);
-        valueOperator.unsafeBufferToIntArrayCopy(dup, srcPosition, dstArray, countInts);
-        end(s);
+    public ByteBuffer getByteBuffer() {
+        ByteBuffer buff = getValueSlice().getByteBuffer();
+        buff = buff.asReadOnlyBuffer();
+        int position = valuePosition();
+        int limit = position + capacity();
+        buff.position(position);
+        buff.limit(limit);
+        return buff.slice();
```

Comment:
Is this where the new object is created? 

In general, I'd suggest quantifying the decision to expose offsets - complexity vs performance. As long as the performance is acceptable (primarily queries) I'd recommend to simplify the programming model and not expose the complexity of working with offsets.